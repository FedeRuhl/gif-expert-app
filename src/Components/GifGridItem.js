import React from 'react';
import PropTypes from 'prop-types';

const GifGridItem = ({ title, url }) => {
    return (
        <div className="card animate_animated animate__fadeIn">
            <h3>{ title }</h3>
            <img src={ url } alt={ title }/>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}

export default GifGridItem