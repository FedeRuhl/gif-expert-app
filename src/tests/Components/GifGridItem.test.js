import '@testing-library/jest-dom';
import { shallow } from 'enzyme'
import GifGridItem from "../../Components/GifGridItem";
import React from 'react';

describe('Gif Grid Item test', function () {
    const title = 'Messi champeon';
    const url = 'https://test.com';
    const wrapper = shallow(<GifGridItem title={ title } url={ url } />);

    test('should show the component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should have a h3 with the title', () => {
        const p = wrapper.find('h3');
        expect(p.text().trim()).toBe(title);
    });

    test('should have a img with the correct url and alt', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });
        
    test('should have a div with the fade in class', () => {
        const div = wrapper.find('div');
        const hasClass = (div.prop('className').includes('animate__fadeIn'));
        expect(hasClass).toBe(true);
    })
    
});