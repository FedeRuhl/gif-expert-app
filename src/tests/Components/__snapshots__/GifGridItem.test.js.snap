// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gif Grid Item test should show the component correctly 1`] = `
<div
  className="card animate_animated animate__fadeIn"
>
  <h3>
    Messi champeon
  </h3>
  <img
    alt="Messi champeon"
    src="https://test.com"
  />
</div>
`;
